File: sdk/samples/WPFSamples/FixedHierarchicalTopologySample/csharp/wizardlauncher.cs
namespace FixedHierarchicalTopologySample
{
    using System;
    using System.Windows.Navigation;
    
    public class WizardLauncher : PageFunction<WizardContext>
    {
        WizardData WizardData = new WizardData();
        
        protected override void Start()
        {
            base.Start();

            // Retain instance in navigation history until Wizard is complete
            this.KeepAlive = true;
            
            // Launch the Wizard
            WizardPage1 WizardPage1 = new WizardPage1(this.WizardData);
            WizardPage1.Return += new ReturnEventHandler<WizardResult>(WizardPage_Return);
            this.NavigationService.Navigate(WizardPage1);
        }

        public void WizardPage_Return(object sender, ReturnEventArgs<WizardResult> e)
        {
            // Wizard was completed (finished or canceled), return WizardResult and WizardData
            OnReturn(new ReturnEventArgs<WizardContext>(new WizardContext(e.Result, this.WizardData)));
        }
    }
}

File: sdk/samples/WPFSamples/FixedHierarchicalTopologySample/visualbasic/wizardlauncher.vb
Imports System
Imports System.Windows.Navigation

Public Class WizardLauncher
    Inherits PageFunction(Of WizardContext)

    Public Sub New()
        Me.WizardData = New WizardData
    End Sub

    Protected Overrides Sub Start()
        MyBase.Start()
        MyBase.KeepAlive = True
        Dim page1 As New WizardPage1(Me.WizardData)
        AddHandler page1.Return, New ReturnEventHandler(Of WizardResult)(AddressOf Me.WizardPage_Return)
        MyBase.NavigationService.Navigate(page1)
    End Sub

    Public Sub WizardPage_Return(ByVal sender As Object, ByVal e As ReturnEventArgs(Of WizardResult))
        Me.OnReturn(New ReturnEventArgs(Of WizardContext)(New WizardContext(e.Result, Me.WizardData)))
    End Sub

    Private WizardData As WizardData

End Class


